dags/atd_service_bot_intake.py

```diff
@@ -0,0 +1,78 @@
+import os
+from datetime import datetime, timedelta
+
+
+from airflow.models import DAG
+from airflow.operators.docker_operator import DockerOperator
+
+from onepasswordconnectsdk.client import Client, new_client
+import onepasswordconnectsdk
+
+DEPLOYMENT_ENVIRONMENT = os.getenv("ENVIRONMENT", "development")
```

Comment:
Thanks @johnclary, that's a really good point.

Right now, the `DEPLOYMENT_ENVIRONMENT` gets its name from being a description of the intent of the particular airflow environment it's running in. Right now we have local ones, `development`, and a production one, `production`. 

I think this is different from different operating modes of a single DAG, which line up really well to the enumeration you stated: `production`, `staging`, & `development`. I think being able to make a DAG kick off in a certain mode is really important, and I don't think our stack is addressing it yet. I think it would be neat to have a DAG check its own tags looking for one that says `operating_mode: staging` or `operating_mode: production` and pick the secrets it needs or to modify its behavior based on that. The Airflow API is available to DAGs during execution, so I am hopeful that this is possible, but I've not tried anything like that yet. 

I'm not sure what is the best, but I'm very much looking forward to more discussion on how we should do this, or if I'm even framing the consideration correctly in my head.

Reply:
I was thinking about this more, and we can totally have a second airflow stack running that has the environment set as "staging" and doing that might allow us to combine the concepts of the airflow deployment environment and the operating mode of DAGs. 🤔